src/java/org/apache/cassandra/utils/concurrent/Ref.java

```diff
@@ -601,6 +601,8 @@ void traverse(final RefCounted.Tidy rootObject)
             InProgressVisit inProgress = null;
             while (inProgress != null || !path.isEmpty())
             {
+                if (Thread.currentThread().isInterrupted())
```

Comment:
this was causing the thread detection check to fail constantly.  Do do `shutdownNow` but since this never checked for interrupted we kept going.

Reply:
Should we re-interrupt here, too?

Reply:
in this case it isn't required as I called `isInterrupted`... I am the caller throwing and checking so I can make sure not to clear the state.

Reply:
your comment did make me double check and ask myself why I didn't interrupt the thread 😉